rasa/core/lock.py

```diff
@@ -0,0 +1,160 @@
+import json
+import logging
+import typing
+from collections import deque
+from typing import Text, Optional, Union, Deque, Dict, Any
+
+import time
+
+if typing.TYPE_CHECKING:
+    pass
+
+logger = logging.getLogger(__name__)
+
+
+class Ticket:
+    def __init__(self, number: int, expires: float):
+        self.number = number
+        self.expires = expires
+
+    def has_expired(self):
+        return time.time() > self.expires
+
+    def dumps(self) -> Text:
+        """Return json dump of `Ticket`."""
+
+        return json.dumps(dict(number=self.number, expires=self.expires))
```

Comment:
I'd actually separate the `as_dict()` stuff cause it makes it easier to reuse and to test, but that's really just a minor comment